### PR TITLE
ARROW-15864: [Java][Docs] Update Arrow nightly Maven releases documentation

### PR DESCRIPTION
**Pull request by davisusanibar**
Current java artifacts nightly build are uploaded to github as an assets.

Update current [documentation](https://github.com/apache/arrow/blob/650f111b524fb1c5bfbfa6f533d15929c90ddc40/docs/source/java/install.rst#installing-nightly-packages) related to how to install java artifacts nightly.

The artifacts are downloaded without problems but .pom and .jar are downloaded as an invalid file format.

Add detail about how to upload jar to maven locally.

### CODE REVIEW

File: docs/source/java/install.rst [hunk 1]
```diff
@@ -83,35 +83,87 @@ Installing Nightly Packages
 
 Arrow nightly builds are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.
 
-To test your code with these artifacts, then configure Maven with:
+Follow the steps and get your nightly packages installed to local maven repository:
```
Comment (lidavidm) [inline docs/source/java/install.rst:86]:
```suggestion
Maven cannot directly use the artifacts from GitHub.
Instead, install them to the local Maven repository:
```

File: docs/source/java/install.rst [hunk 2]
```diff
@@ -83,35 +83,87 @@ Installing Nightly Packages
...
+
+    $ mkdir nightly-2022-03-03-0-github-java-jars
+    $ cd nightly-2022-03-03-0-github-java-jars
+    $ wget https://github.com/ursacomputing/crossbow/releases/download/nightly-2022-03-03-0-github-java-jars/arrow-vector-8.0.0.dev165.jar
+    $ wget https://github.com/ursacomputing/crossbow/releases/download/nightly-2022-03-03-0-github-java-jars/arrow-format-8.0.0.dev165.jar
+    $ tree
+    |__ arrow-format-8.0.0.dev165.jar
+    |__ arrow-vector-8.0.0.dev165.jar
+
+4. Install nightly java version to local maven repository with `mvn install:install-file`
```
Comment (lidavidm) [inline docs/source/java/install.rst:124]:
```suggestion
4. Install the artifacts to the local Maven repository with ``mvn install:install-file``
```

File: docs/source/java/install.rst [hunk 3]
```diff
@@ -83,35 +83,87 @@ Installing Nightly Packages
...
+        -Dfile="$(pwd)/arrow-vector-8.0.0.dev165.jar" \
+        -DgroupId=org.apache.arrow \
+        -DartifactId=arrow-vector \
+        -Dversion=8.0.0.dev165 \
+        -Dpackaging=jar \
+        -DcreateChecksum=true \
+        -Dgenerate.pom=true
+    [INFO] Installing /nightly-2022-03-03-0-github-java-jars/arrow-vector-8.0.0.dev165.jar to /Users/arrow/.m2/repository/org/apache/arrow/arrow-vector/8.0.0.dev165/arrow-vector-8.0.0.dev165.jar
+
+6. Validate packages installed locally on maven repository:
```
Comment (lidavidm) [inline docs/source/java/install.rst:147]:
```suggestion
6. Validate that the packages were installed:
```

File: docs/source/java/install.rst [hunk 4]
```diff
@@ -83,35 +83,87 @@ Installing Nightly Packages
...
+            |__ arrow-vector-8.0.0.dev165.pom
+
+5. Compile your project with `mvn clean install`.
+
+.. code-block:: shell
+
+    $ mvn clean install
+    [INFO] BUILD SUCCESS
 
 Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
```
Comment (lidavidm) [inline docs/source/java/install.rst:168]: This should go up at the very top as part of the paragraph about GitHub. 

```
Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.
```

Comment (davisusanibar) [reply docs/source/java/install.rst]: Added to the initial part of: Installing Nightly Packages

File: docs/source/java/install.rst [hunk 5]
```diff
@@ -81,39 +81,91 @@ Installing Nightly Packages
...
+        -Dfile="$(pwd)/arrow-vector-8.0.0.dev165.jar" \
+        -DgroupId=org.apache.arrow \
+        -DartifactId=arrow-vector \
+        -Dversion=8.0.0.dev165 \
+        -Dpackaging=jar \
+        -DcreateChecksum=true \
+        -Dgenerate.pom=true
+    [INFO] Installing /nightly-2022-03-03-0-github-java-jars/arrow-vector-8.0.0.dev165.jar to /Users/arrow/.m2/repository/org/apache/arrow/arrow-vector/8.0.0.dev165/arrow-vector-8.0.0.dev165.jar
+
+6. Validate that the packages were installed:
```
Comment (lidavidm) [inline docs/source/java/install.rst:149]: The numbering is off (it goes 6 then 5)

File: docs/source/java/install.rst [hunk 6]
```diff
@@ -81,39 +81,91 @@ Installing Nightly Packages
...
+        |__ 8.0.0.dev165
+            |__ arrow-vector-8.0.0.dev165.jar
+            |__ arrow-vector-8.0.0.dev165.pom
+
+5. Compile your project with `mvn clean install`.
+
+.. code-block:: shell
+
+    $ mvn clean install
+    [INFO] BUILD SUCCESS
```
Comment (lidavidm) [inline docs/source/java/install.rst:168]: I don't think we need to show this, just say something like 

```
Compile your project like usual with ``mvn install``.
```

Comment (davisusanibar) [reply docs/source/java/install.rst]: Deleted

File: docs/source/java/install.rst (hunk 5 above)
Comment (davisusanibar) [reply docs/source/java/install.rst]: Changed

File: docs/source/java/install.rst [hunk 7]
```diff
@@ -81,39 +81,86 @@ Installing Nightly Packages
...
+Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
+The artifacts are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.
 
-To test your code with these artifacts, then configure Maven with:
+Maven cannot directly use the artifacts from GitHub.
+Instead, install them to the local Maven repository:
 
-.. code-block:: xml
+1. Decide nightly packages repository to use, for example: https://github.com/ursacomputing/crossbow/releases/tag/nightly-2022-03-03-0-github-java-jars
+2. Define nightly packages to use, for example: `arrow-vector` and `arrow-format`
```
Comment (lidavidm) [inline docs/source/java/install.rst:91]:
```suggestion
2. Add packages to your pom.xml, for example: ``arrow-vector`` and ``arrow-format``:
```

File: docs/source/java/install.rst [hunk 8]
```diff
@@ -81,39 +81,86 @@ Installing Nightly Packages
...
+            <version>${arrow.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-format</artifactId>
+            <version>${arrow.version}</version>
+        </dependency>
+    </dependencies>
+
+3. Download packages needed to a temporary directory
```
Comment (lidavidm) [inline docs/source/java/install.rst:114]:
```suggestion
3. Download packages needed to a temporary directory:
```

File: docs/source/java/install.rst [hunk 9]
```diff
@@ -81,39 +81,86 @@ Installing Nightly Packages
...
+
+    $ mkdir nightly-2022-03-03-0-github-java-jars
+    $ cd nightly-2022-03-03-0-github-java-jars
+    $ wget https://github.com/ursacomputing/crossbow/releases/download/nightly-2022-03-03-0-github-java-jars/arrow-vector-8.0.0.dev165.jar
+    $ wget https://github.com/ursacomputing/crossbow/releases/download/nightly-2022-03-03-0-github-java-jars/arrow-format-8.0.0.dev165.jar
+    $ tree
+    |__ arrow-format-8.0.0.dev165.jar
+    |__ arrow-vector-8.0.0.dev165.jar
+
+4. Install the artifacts to the local Maven repository with ``mvn install:install-file``
```
Comment (lidavidm) [inline docs/source/java/install.rst:126]:
```suggestion
4. Install the artifacts to the local Maven repository with ``mvn install:install-file``:
```

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
+1, will merge after CI

**Pull request closed without merging**